pkg/tls/certificate.go

```diff
@@ -185,24 +136,27 @@ func (c *Certificate) AppendCertificate(certs map[string]map[string]*tls.Certifi
 			}
 		}
 	}
+
+	// Guarantees the order to produce a unique cert key.
+	sort.Strings(SANs)
```

Comment:
Considering this array also includes the CN of the cert, `SANs` does not sound like a correct name.